### PR TITLE
updates to security guidelines and docker config

### PR DESCRIPTION
**Pull request by agunapal**
## Description

Update Security guidelines for using TorchServe

Fixes #(issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

Please describe the Unit or Integration tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
Logs for Test A

- [ ] Test B
Logs for Test B

## Checklist:

- [ ] Did you have fun?
- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -14,6 +14,19 @@ TorchServe as much as possible relies on automated tools to do security scanning
...
+## Important Security Guidelines
+
+1. TorchServe listens on the following ports
+    1. HTTP - `8080`, `8081`, `8082`
+    2. gRPC - `7070`, `7071`
+
+    These ports are accessible to `localhost` by default.  The address can be configured by following the [guide](https://github.com/pytorch/serve/blob/master/docs/configuration.md#configure-torchserve-listening-address-and-port)
+    TorchServe does not prevent users from configuring the address to be `0.0.0.0`. Please be aware of the security risks if you use `0.0.0.0`
+2. TorchServe's Docker image is conigured to listen to `localhost` by [default](https://github.com/pytorch/serve/blob/master/docker/config.properties)
+
```
Comment (msaroufim) [inline SECURITY.md:26]: also mention that torchserve can run arbitrary python files so don't download mar files you don't trust from the internet

File: .github/workflows/regression_tests_docker.yml [hunk 2]
```diff
@@ -1,9 +1,10 @@
 name: Run Regression Tests on Docker
 
 on:
+  push
   # run every day at 5:15am
-  schedule:
-    - cron:  '15 5 * * *'
+  #schedule:
+  #  - cron:  '15 5 * * *'
```
[the remark this answers is not in the document]
Comment (agunapal) [reply .github/workflows/regression_tests_docker.yml:7]: Yes, i saw this when I was trying how to auto-retry workflows. They use the same approach to do this.

File: SECURITY.md (hunk 1 above)
Comment (agunapal) [reply SECURITY.md:26]: done

File: .github/workflows/regression_tests_docker.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (agunapal) [reply .github/workflows/regression_tests_docker.yml]: closing this as 0.0.0.0 is the right value and not localhost

File: SECURITY.md (hunk 1 above)
Comment (msaroufim) [reply SECURITY.md:28]: the updated language is good but please explicitly also mention that torchserve executes arbitrary python files when running a mar file, the risk is not just in downloading

Comment (agunapal) [reply SECURITY.md:28]: Got it. So, how do they verify it? We should mention that too

Comment (msaroufim) [reply SECURITY.md:30]: There is no generic way to verify if a piece of python code is safe, they key is that it needs to be code from a source they trust

File: SECURITY.md [hunk 3]
```diff
@@ -14,6 +14,23 @@ TorchServe as much as possible relies on automated tools to do security scanning
...
+    1. HTTP - `8080`, `8081`, `8082`
+    2. gRPC - `7070`, `7071`
+
+    These ports are accessible to `localhost` by default.  The address can be configured by following the [guide](https://github.com/pytorch/serve/blob/master/docs/configuration.md#configure-torchserve-listening-address-and-port)
+    TorchServe does not prevent users from configuring the address to be `0.0.0.0`. Please be aware of the security risks if you use `0.0.0.0`
+2. TorchServe's Docker image is configured to expose the ports `8080`, `8081`, `8082`, `7070`, `7071` to the host by default(https://github.com/pytorch/serve/blob/master/docker/Dockerfile). When these ports are mapped to the host, make sure to use `localhost` or a specific ip address.
+
+3. Be sure to validate the authenticity of the `.mar` file being used with TorchServe.
+    1. A `.mar` file being downloaded from the internet from an untrusted source may have malicious code, compromising the integrity of your application
+    2. TorchServe executes arbitrary python code packaged in the `mar` file. Make sure the code being packaged is not creating a security risk.
```
Comment (msaroufim) [inline SECURITY.md:29]:
```suggestion
    2. TorchServe executes arbitrary python code packaged in the `mar` file. Make sure that you've either audited that the code you're using is safe and/or is from a source that you trust
```

**Review by msaroufim — APPROVED** [review]

**Review by lxning — APPROVED** [review]

File: SECURITY.md [hunk 4]
```diff
@@ -14,6 +14,23 @@ TorchServe as much as possible relies on automated tools to do security scanning
...
+    2. gRPC - `7070`, `7071`
+
+    These ports are accessible to `localhost` by default.  The address can be configured by following the [guide](https://github.com/pytorch/serve/blob/master/docs/configuration.md#configure-torchserve-listening-address-and-port)
+    TorchServe does not prevent users from configuring the address to be `0.0.0.0`. Please be aware of the security risks if you use `0.0.0.0`
+2. TorchServe's Docker image is configured to expose the ports `8080`, `8081`, `8082`, `7070`, `7071` to the host by default(https://github.com/pytorch/serve/blob/master/docker/Dockerfile). When these ports are mapped to the host, make sure to use `localhost` or a specific ip address.
+
+3. Be sure to validate the authenticity of the `.mar` file being used with TorchServe.
+    1. A `.mar` file being downloaded from the internet from an untrusted source may have malicious code, compromising the integrity of your application
+    2. TorchServe executes arbitrary python code packaged in the `mar` file. Make sure that you've either audited that the code you're using is safe and/or is from a source that you trust
+
```
Comment (lxning) [inline SECURITY.md:30]: archiver can support mar, tgz and folder. we can make it more generic at this section to let cx know they are responsible for the security of the code in model artifacts

**Pull request merged**